### ISSUE TITLE
Please reopen# 3634

### ISSUE THREAD

**Issue by machineghost** [styled-components/styled-components#3921]
## Environment

`npx envinfo --system --binaries --npmPackages styled-components,babel-plugin-styled-components --markdown --clipboard`

## Reproduction

https://codesandbox.io/s/styled-components-base-gc367

## Actual Behavior
Please reopen ticket #3634.  It was closed prematurely (as evidenced by the fact that the last comment says as much, with 8 upvotes).

**Comment by probablyup** [comment]
Hi there, someone from the Next team reached out and I gave feedback for updating their examples. I believe the issue with the new Next app directory setup is they were hacking StyleSheetManager instead of just allowing for typical rehydration after SSR

**Comment by machineghost** [comment]
Well, whatever you told them, or whatever they implemented, someone got something wrong, as the issue remains for many users (at least 9, per that ticket).

**Comment by probablyup** [comment]
Yeah I have no control over them updating their examples and repos. Everything works fine with the previous setup, so the issue is on their end not ours

**Comment by machineghost** [comment]
Just FYI, you're literally using Styled Components users over this; see the most recent comments (plural) on: https://github.com/vercel/next.js/issues/30802#issuecomment-1547904733
